components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useRef } from "react";
+import Icon from "../Icon/Icon";
+import "./ProgressBar.scss";
+
+type Props = {
+  value: number;
+  onChange: (newProgress: number) => void;
+};
+
+const ProgressBar: React.FC<Props> = ({ value, onChange }) => {
+  const [isDragging, setIsDragging] = useState(false);
+  const [dragStartX, setDragStartX] = useState(0);
+  const [dragStartValue, setDragStartValue] = useState(value);
+  const progressBarRef = useRef<HTMLDivElement>(null);
+
+  const handleIconMouseDown = (event: React.MouseEvent<HTMLDivElement>) => {
+    event.stopPropagation();
+    setIsDragging(true);
+    setDragStartX(event.clientX);
+    setDragStartValue(value);
+  };
+
+  const calculatePinPosition = (event: React.MouseEvent<HTMLDivElement>) => {
+    const progressBarRect = progressBarRef.current?.getBoundingClientRect();
+    const progressBarWidth = progressBarRect?.width || 0;
```

Comment:
If we divide by zero, we get `Infinity`. Not sure we wanted infinity progress bar width 😅